internal/servers/server/option_test.go

```diff
@@ -0,0 +1,1673 @@
+package server
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/vdaas/vald/internal/errgroup"
+	"github.com/vdaas/vald/internal/errors"
+	"google.golang.org/grpc"
+)
+
+func TestWithHost(t *testing.T) {
+	type test struct {
+		name      string
+		host      string
+		checkFunc func(opt Option) error
+	}
+
+	tests := []test{
+		{
+			name: "set success",
+			host: "host",
+			checkFunc: func(opt Option) error {
+				got := new(server)
+				opt(got)
+
+				if got.host != "host" {
+					return errors.New("invalid param was set")
+				}
+				return nil
+			},
+		},
+
+		{
+			name: "do nothing",
+			host: "",
+			checkFunc: func(opt Option) error {
+				got := new(server)
+				opt(got)
+
+				if got.host != "" {
+					return errors.New("invalid param was set")
+				}
+				return nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			opt := WithHost(tt.host)
+			if err := tt.checkFunc(opt); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestWithPort(t *testing.T) {
+	type test struct {
+		name      string
+		port      uint
+		checkFunc func(opt Option) error
+	}
+
+	tests := []test{
+		{
+			name: "set success",
+			port: 8080,
+			checkFunc: func(opt Option) error {
+				got := new(server)
+				opt(got)
+
+				if got.port != 8080 {
+					return errors.New("invalid param was set")
+				}
+				return nil
+			},
+		},
+
+		{
+			name: "do nothing",
+			checkFunc: func(opt Option) error {
+				got := new(server)
+				opt(got)
+
+				if got.port != 0 {
+					return errors.New("invalid param was set")
+				}
+				return nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			opt := WithPort(tt.port)
+			if err := tt.checkFunc(opt); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestWithName(t *testing.T) {
+	type test struct {
+		name      string
+		arg       string
+		checkFunc func(opt Option) error
+	}
+
+	tests := []test{
+		{
+			name: "set success",
+			arg:  "vald",
+			checkFunc: func(opt Option) error {
+				got := new(server)
+				opt(got)
+
+				if got.name != "vald" {
+					return errors.New("invalid param was set")
+				}
+				return nil
+			},
+		},
```

Comment:
- Could you please add test whose name(param of `WithName`) is default value ?

Reply:
I need not to check the default value because I just set the argument to the field.:thinking:

```
func WithName(name string) Option {
	return func(s *server) {
		s.name = name
	}
}

```

Reply:
you should check argument variable before set to field variable, beacuse usual FOP has default Options if argument is empty or nil param default option will be flushed by empty set

Reply:
oh, I understood.
thank you for review.